config/routes.rb

```diff
@@ -8,8 +8,8 @@
     post "datatable", on: :collection
   end
 
-  post "table_state_update", to: "preference_sets#table_state_update"
-  get "table_state", to: "preference_sets#table_state"
+  post "/table_state_update/:table_name", to: "preference_sets#table_state_update", as: :table_state_update
+  get "/table_state/:table_name", to: "preference_sets#table_state", as: :table_state
```

Comment:
follow the same convention that casa is currently doing. This should look something like
```
resources :preference_sets, only: [] do
  member do
   post table_state_updated
   get table_state
  end
end
```

Reply:
make sense!

Reply:
sounds good.  
How do you like this:
`  resources :preference_sets, only: [] do
    collection do
      post "/table_state_update/:table_name", to: "preference_sets#table_state_update", as: :table_state_update
      get "/table_state/:table_name", to: "preference_sets#table_state", as: :table_state
    end
  end`

I used the collection helper because the member do requires the parent's id. 
Also, i'm reading to see if I can find a better way to write the nested routes with helpers